### ISSUE TITLE
Temporary false positives after saving file when Microsoft Python Extension is enabled

### ISSUE THREAD

**Issue by adonig** [microsoft/pyright#4337]
**Describe the bug**
Pyright reports false positives after saving a Python file when the Microsoft Python Extension is enabled. They go away around 7 seconds after they appeared.

**To Reproduce**
Install and enable the pyright extension. Install and enable the Microsoft Python Extension. Set the language server to None. Save the code below.

**Expected behavior**
No false positives. The same behavior like when the Microsoft Python Extension is disabled.

**Screenshots or Code**

```python
# pyright: strict

from typing import Any

from aws_cdk import Stack
from aws_cdk import aws_ec2 as ec2
from constructs import Construct


class HelloCdkStack(Stack):
    def __init__(self, scope: Construct, construct_id: str, **kwargs: Any) -> None:
        super().__init__(scope, construct_id, **kwargs)

        vpc = ec2.Vpc(self, "VPC")
```
![Screenshot from 2022-12-14 17-25-38](https://user-images.githubusercontent.com/900404/207661027-4c5390fd-2f4d-4e19-8b74-758aeb170915.png)

When the Microsoft Python extension is disabled, everything is fine:

![Screenshot from 2022-12-14 17-20-23](https://user-images.githubusercontent.com/900404/207661305-e20bf7ae-3171-4f19-972a-10a748170eb8.png)

![Screenshot from 2022-12-14 17-20-39](https://user-images.githubusercontent.com/900404/207661311-e3cec309-9854-4113-ba46-76686359d088.png)

Install the following dependencies:

```toml
[tool.poetry.dependencies]
python = "^3.11"
aws-cdk-lib = "2.54.0"
constructs = ">=10.0.0,<11.0.0"
```

**VS Code extension or command-line**
I use the pyright extension version v1.1.284 and the Microsoft Python extension version v2022.20.1

**Additional context**
I use VSCodium 1.74.0 on Fedora Linux 37 because I can´t use proprietary software (OS X, VSCode, pylance) in this environment.

When I run pyright on the command line if reports no errors either:

```
$ pyright
No configuration file found.
pyproject.toml file found at /home/asd/Code/hello/cdk.
Loading pyproject.toml file at /home/asd/Code/hello/cdk/pyproject.toml
Assuming Python version 3.11
Assuming Python platform Linux
No include entries specified; assuming /home/asd/Code/hello/cdk
stubPath /home/asd/Code/hello/cdk/typings is not a valid directory.
Searching for source files
Found 6 source files
pyright 1.1.284
0 errors, 0 warnings, 0 informations 
Completed in 4.345sec
```

**Comment by erictraut** [comment]
I'm not able to repro this on my Mac with VS Code. I don't have a Linux machine, so I can't recreate your environment precisely.

Are you saying that the error doesn't occur prior to saving, then you save the file and the error appears, and then seven seconds later the error disappears?

The only interaction between the Python and Pyright extensions is that the Python extension provides the selected path of the selected Python environment. You can also manually set this using the `analysis.pythonPath` setting. Please try setting that to see if it alters the behavior. I can't think of any other theories as to why you're seeing this behavior.

We test the Python and Pyright extensions with VS Code. Vscodium isn't a supported product, so you may need to debug this one yourself. If you find the cause of the problem, please let me know and I'll work with you to incorporate a fix.

**Comment by adonig** [comment]
> Are you saying that the error doesn't occur prior to saving, then you save the file and the error appears, and then seven seconds later the error disappears?

Exactly. I made a short screen recording:

[Screencast from 2022-12-15 10-13-56.webm](https://user-images.githubusercontent.com/900404/207820190-2d864fd7-d18a-41a5-82fe-2778edea1d54.webm)

The .venv is in my project folder so I had to put this in pyproject.toml to make pyright work. Maybe it's somehow related?

```toml
[tool.pyright]
exclude = [ ".venv" ]
venvPath = "."
venv = ".venv"
```

**Comment by adonig** [comment]
I found out that it can happen when the Microsoft Python extension is disabled, but it is very hard to reproduce. When the Microsoft Python extension is enabled though, it happens all the time. 

[Screencast from 2022-12-15 12-40-33.webm](https://user-images.githubusercontent.com/900404/207851027-c185de8a-d1b5-42df-84e5-a645b02935bb.webm)
